src/sql/platform/accounts/browser/firewallRuleDialogController.ts

```diff
@@ -57,31 +57,28 @@ export class FirewallRuleDialogController {
 		this._errorMessageService.showDialog(Severity.Error, this._addAccountErrorTitle, message);
 	}
 
-	private handleOnCreateFirewallRule(): void {
+	private async handleOnCreateFirewallRule(): Promise<void> {
 		const resourceProviderId = this._resourceProviderId;
-
-		this._accountManagementService.getSecurityToken(this._firewallRuleDialog.viewModel.selectedAccount, AzureResource.ResourceManagement).then(tokenMappings => {
-			let firewallRuleInfo: azdata.FirewallRuleInfo = {
+		try {
+			const securityTokenMappings = await this._accountManagementService.getSecurityToken(this._firewallRuleDialog.viewModel.selectedAccount!, AzureResource.ResourceManagement);
```

Comment:
For some reason typescript wasn't resolving the type here correctly when using promises but did when using await *shrug*